content/HTTPS is not enough - Pablo Blanco.md

```diff
@@ -0,0 +1,218 @@
+insecure website using a man-in-the-middle attack. This is not as rare as it
+seems – it just means that the site is not present in the browser’s HSTS
+database. While this can be the result of a time-based attack via NTP, it can
+also happen if the max-age expiry period since the last visit has passed, the
+browser is a fresh install, the server is misconfigured (for example max-age was
+left at 0 after testing) and so on.
+
+To remedy this issue and also improve page load speeds,
+[modern browsers](https://caniuse.com/#feat=stricttransportsecurity) (Chrome,
+Firefox, Safari, among others) check if a certain domain is HSTS-enabled against
+a common preloaded list called [hstspreload.org](https://hstspreload.org/).
+
+There, webmasters and site administrators can submit it’s domains for inclusion
+in Chrome's HTTP Strict Transport Security (HSTS) preload list. By including our
+domains to that list we are sure to force the user’s browser to navigate our
+site over HTTPS **strictly**.
+
+If a site is listed, the browser always connects to it using HTTPS, even on the
+first visit. This effectively eliminates the only significant vulnerability in
+HTTPS enforcement, ensuring end-to-end encryption.
+
+**To add our site to the preloaded list we need:**
+
+1. Present a valid certificate
+2. Redirect all HTTP requests on port 80 to HTTPS
+3. Include the HSTS header in any additional HTTPS redirects
+4. Serve all subdomains over HTTPS
+5. Specify all header directives: max-age must be at least 31536000 seconds (1
+   year), and includesubdomains and preload must be present
+
+### Content Security Policy (CSP)
+
+Another header that can help us is Content Security Policy (CSP), this is used
+to detect and mitigate certain types of attacks like cross-site scripting (XSS),
+clickjacking and other code injection attacks resulting from execution of
+malicious content in the trusted web page context. A CSP can be used for simple
+purposes like enforcing HTTPS on SSL-enabled sites, to more sophisticated uses
+like authorizing only truly trusted sources and blocking others.
+
+Using CSP also makes sure that we don’t have mixed content errors on your site.
+For example, if we have a HTTPS site and HTTP is embedded in the site (like a
+video), then we will wind up with mixed content errors in the browser console. A
+CSP could direct a website to make all URLs are always HTTPS, to avoid any mixed
+content errors.
+
+This header contains only the policy, and can be defined this way:
+
+`Content-Security-Policy: default-src 'self'`
+
+Here we are saying that all content to come from the site's own origin
+(excluding subdomains.) The policy provides many options that can be found
+[here](https://developer.mozilla.org/en-US/docs/Web/HTTP/CSP).
+
+**How can I check if any site uses security headers?**
+
+We can check server response headers by using curl or nmap (more detailed
+information), here a few examples:
```

Comment:
```suggestion
information), here are a few examples:
```

Reply:
We can check server response headers by using curl or nmap _for more detailed information_.